bigquery/google/cloud/bigquery/schema.py

```diff
@@ -32,21 +32,34 @@ class SchemaField(object):
     :type description: str
     :param description: optional description for the field.
 
-    :type fields: list of :class:`SchemaField`, or None
+    :type fields: tuple of :class:`SchemaField`
     :param fields: subfields (requires ``field_type`` of 'RECORD').
     """
     def __init__(self, name, field_type, mode='NULLABLE', description=None,
-                 fields=None):
+                 fields=()):
         self.name = name
         self.field_type = field_type
         self.mode = mode
         self.description = description
-        self.fields = fields
+        self.fields = tuple(fields)
 
-    def __eq__(self, other):
+    def _key(self):
+        """
+        A tuple describing the contents of this :class:`SchemaField`.
+        Used to compute this instance's hashcode and evaluate equality.
+        """
         return (
-            self.name == other.name and
-            self.field_type.lower() == other.field_type.lower() and
-            self.mode == other.mode and
-            self.description == other.description and
-            self.fields == other.fields)
+            self.name,
+            self.field_type.lower(),
+            self.mode,
+            self.description,
+            self.fields)
+
+    def __eq__(self, other):
+        return self._key() == other._key()
+
+    def __hash__(self):
+        return hash(self._key())
+
+    def __repr__(self):
```

Comment:
@dhermes @tseaver thought that I might as well add this given that I'm touching this class